File: app/src/main/java/co/early/password123/ui/common/SyncerAnimationComplete.java
package co.early.password123.ui.common;


import android.animation.Animator;
import android.os.Handler;
import android.view.View;

import co.early.asaf.core.Affirm;
import co.early.asaf.core.ui.SyncableView;


public class SyncerAnimationComplete implements Animator.AnimatorListener {

    private final SyncableView syncableView;
    private final View[] viewsToReset;


    public SyncerAnimationComplete(SyncableView syncableView, View... viewsToReset) {
        this.syncableView = Affirm.notNull(syncableView);
        this.viewsToReset = Affirm.notNull(viewsToReset);
    }

    public SyncerAnimationComplete(SyncableView syncableView) {
        this.syncableView = Affirm.notNull(syncableView);
        this.viewsToReset = new View[0];
    }



    @Override
    public void onAnimationStart(Animator animation) {

    }

    @Override
    public void onAnimationEnd(Animator animation) {

        //otherwise when we draw, the animation is still "running" on older android
        new Handler().postDelayed(() -> {

            for (View view : viewsToReset) {
                view.setAlpha(1.0f);
                view.setRotation(0f);
            }
            syncableView.syncView();

        }, 10);//

    }

    @Override
    public void onAnimationCancel(Animator animation) {

    }

    @Override
    public void onAnimationRepeat(Animator animation) {

    }
}

File: app/src/main/java/co/early/password123/ui/common/anim/SyncerAnimationComplete.kt
package co.early.password123.ui.common.anim


import android.animation.Animator
import android.os.Handler
import android.view.View

import co.early.asaf.core.Affirm
import co.early.asaf.core.ui.SyncableView


class SyncerAnimationComplete : Animator.AnimatorListener {

    private val syncableView: SyncableView
    private val viewsToReset: Array<out View>


    constructor(syncableView: SyncableView, vararg viewsToReset: View) {
        this.syncableView = Affirm.notNull(syncableView)
        this.viewsToReset = Affirm.notNull(viewsToReset)
    }

    constructor(syncableView: SyncableView) {
        this.syncableView = Affirm.notNull(syncableView)
        this.viewsToReset = emptyArray()
    }


    override fun onAnimationStart(animation: Animator) {

    }

    override fun onAnimationEnd(animation: Animator) {

        //otherwise when we draw, the animation is still "running" on older android
        Handler().postDelayed({

            for (view in viewsToReset) {
                view.alpha = 1.0f
                view.rotation = 0f
            }
            syncableView.syncView()

        }, 10)//

    }

    override fun onAnimationCancel(animation: Animator) {

    }

    override fun onAnimationRepeat(animation: Animator) {

    }
}
